lib/poseidon/sync_producer.rb

```diff
@@ -57,6 +57,7 @@ def send_messages(messages)
         if !messages_to_send.pending_messages? || @max_send_retries == 0
           break
         else
+          Poseidon.logger.debug { "retrying sending #{messages_to_send.messages} messages"  }
```

Comment:
Interesting. Haven't seen this pattern before but matching the rest of the project seems good